src/interpreter/MarkdownParser.js

```diff
@@ -4,48 +4,124 @@ const TestSuiteDescriptor = require('../testsuite/TestSuiteDescriptor');
 const path = require('path');
 
 const fs = require('fs'); // eslint-disable-line no-unused-vars
-const descriptorFlags_re = /\[\*(.*)\].*/ig;
-const resultFlags_re = /\[&(.*)\].*/ig;
-const inside_parenthesis_re = /\((.*)\)/;
+const DESCRIPTOR_FLAGS_REGEX = /\[\*(.*)\].*/ig;
+const INSIDE_PARENTHESIS_REGEX = /\((.*)\)/;
+const TEST_DETECTION_REGEX = /\[.*?\]\(\?=.*?\)\)/gm;
+const TEST_PARAMETER_NAME_REGEX = /[\w\d]+/gm;
+
+const PARAMETER_DETECTION_REGEX = /\[.*?\]\(#.*?\)/gm;
+
+const TEST_EXPECTED_RESULT_INDEX = 1;
+const TEST_CLASS_INDEX = 2;
+const TEST_NAME_INDEX = 3;
+const TEST_PARAMETERS_INDEX = 4;
+
+const PARAMETER_VALUE_INDEX = 1;
+const PARAMETER_NAME_INDEX = 2;
+
 
 class MarkdownParser extends GenericParser{
     parseFile(filePath, options){ // eslint-disable-line no-unused-vars
         let mdContent = fs.readFileSync(filePath, 'UTF-8');
-        let descriptor_tags = this._getTags(mdContent, descriptorFlags_re);
-        let descriptor = this._generateDescriptor(descriptor_tags, filePath);
-        let result_tags = this._getTags(mdContent, resultFlags_re);
-        let tests = this._generateTests(result_tags);
 
-        descriptor.setTests(tests);
-        return descriptor;
+        let tests = this._generateTests(mdContent);
+
+        return this._generateDescriptor(mdContent, filePath, tests);
     }
 
-    _getTags(mdContent, re) {
+    _getDescriptorTags(mdContent) {
         let tags = [];
         let matches;
-        while ((matches = re.exec(mdContent)) !== null) {
-            tags[matches[1]] = inside_parenthesis_re.exec(matches[0])[1];
+        while ((matches = DESCRIPTOR_FLAGS_REGEX.exec(mdContent)) !== null) {
+            tags[matches[1]] = INSIDE_PARENTHESIS_REGEX.exec(matches[0])[1];
         }
         return tags;
     }
```

Comment:
This might be problematic on the second execution.
Regex objects return null on their second execution if they contain a 'g' flag.
new RegExp(regex[, flags]); prevents this.

Reply:
fixed